### PR TITLE
Ndrcn 275

### PR DESCRIPTION
**Pull request by somaraja**

### CODE REVIEW

File: app/models/UserDetails.scala [hunk 1]
```diff
@@ -19,7 +19,6 @@ package models
 import play.api.libs.json.{Json, OFormat}
 
 final case class UserDetails(
-                             VATNumber: Option[VRN],
```
Comment (lmugnai) [inline app/models/UserDetails.scala:22]: @somaraja can we put isVatRegistered in here?

File: app/models/requests/CreateClaimRequest.scala [hunk 2]
```diff
@@ -65,7 +65,6 @@ object CreateClaimRequest {
       telephone <- userAnswers.get(PhoneNumberPage)
       email <-userAnswers.get(EmailAddressPage)
     } yield UserDetails(
-      None,
```
Comment (lmugnai) [inline app/models/requests/CreateClaimRequest.scala:68]: can we put isVatRegistered here?

**Review by lmugnai — CHANGES_REQUESTED** [review]

File: app/models/UserDetails.scala (hunk 1 above)
Comment (somaraja) [reply app/models/UserDetails.scala:22]: @lmugnai this is incorporated

File: app/controllers/EntryDetailsController.scala [hunk 3]
```diff
@@ -47,6 +47,10 @@ class EntryDetailsController @Inject()(
   def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
 
+      if(request.userAnswers.get(CustomsRegulationTypePage) == Some(CustomsRegulationType.UnionsCustomsCodeRegulation)) {
```
Comment (lmugnai) [inline app/controllers/EntryDetailsController.scala:50]: Sorry, just spotted this and giving it some thought, I'm not sure if this is the right place to set this answer. Could we not set the user answers on submit of CustomsRegluationTypePage? Also, I think the ArticleType should be set to Schedule if the user selects UKCustomsCodeRegulation.

File: app/navigation/Navigator.scala [hunk 4]
```diff
@@ -63,6 +63,11 @@ class Navigator @Inject()() {
     case _ => _ => routes.IndexController.onPageLoad()
   }
 
+  private def getEntryDetails(answers: UserAnswers): Call = answers.get(CustomsRegulationTypePage) match {
```
Comment (lmugnai) [inline app/navigation/Navigator.scala:66]: I think this is the wrong way round. Want to have a chat about it?

**Review by lmugnai — CHANGES_REQUESTED** [review]

File: app/controllers/EntryDetailsController.scala (hunk 3 above)
Comment (somaraja) [reply app/controllers/EntryDetailsController.scala]: This review comment is incorporated.

File: app/navigation/Navigator.scala (hunk 4 above)
Comment (somaraja) [reply app/navigation/Navigator.scala:66]: This review comment is incorporated.

**Pull request merged**